### PR TITLE
Cleaning up

### PR DESCRIPTION
**Pull request by philips77**
Minor polishing and warning suppression.

One meaningful change: changing `BleDevice.name` to optional.

### CODE REVIEW

File: core/src/main/java/no/nordicsemi/android/kotlin/ble/core/BleDevice.kt [hunk 1]
```diff
@@ -75,16 +75,16 @@ sealed interface BleDevice : Parcelable {
...
      */
     val hasName
-        get() = name.isNotEmpty()
+        get() = name?.isNotEmpty() == true
 }
 
 /**
  * Class representing BLE server device. It can be either mocked or native variant.
- * It can be connected to using [ClientBleGatt].
+ * It can be connected to using [ClientDevice].
```
Comment (sylwester-zielinski) [inline core/src/main/java/no/nordicsemi/android/kotlin/ble/core/BleDevice.kt:83]: ClientBleGatt is used to create connections

File: client-android/src/main/java/no/nordicsemi/android/kotlin/ble/client/real/NativeClientBleAPI.kt [hunk 2]
```diff
@@ -84,55 +86,55 @@ class NativeClientBleAPI(
         value: DataByteArray,
         writeType: BleWriteType
     ) {
-        val characteristic = (characteristic as NativeBluetoothGattCharacteristic).characteristic
+        val c = (characteristic as NativeBluetoothGattCharacteristic).characteristic
```
Comment (sylwester-zielinski) [inline client-android/src/main/java/no/nordicsemi/android/kotlin/ble/client/real/NativeClientBleAPI.kt:89]: Why such a short name? If the code is long enough you will have to look what this variable means.

File: core/src/main/java/no/nordicsemi/android/kotlin/ble/core/BleDevice.kt (hunk 1 above)
Comment (philips77) [reply core/src/main/java/no/nordicsemi/android/kotlin/ble/core/BleDevice.kt:83]: Sorry, it's jus not visible from that module and AS shows as warning, at least for me.

File: client-android/src/main/java/no/nordicsemi/android/kotlin/ble/client/real/NativeClientBleAPI.kt (hunk 2 above)
Comment (philips77) [reply client-android/src/main/java/no/nordicsemi/android/kotlin/ble/client/real/NativeClientBleAPI.kt:89]: There was a warning, that the new `characteristic` hides the method parameter.
I agree, that `c` is a bad name. I'l revert.

**Review by sylwester-zielinski — APPROVED** [review]
Amazing!

**Pull request merged**